exonum/src/storage/proof_list_index/mod.rs

```diff
@@ -447,7 +476,7 @@ where
 
 impl<'a, V> ProofListIndex<&'a mut Fork, V>
 where
-    V: StorageValue,
+    V: StorageValue + Clone,
```

Comment:
That's a significant change; is it really justified? If `HashStream` worked ideally, I'd suggest to extend `CryptoHash` to

```rust
trait CryptoHash {
    fn hash_to(&self, state: HashStream) -> HashStream;
    // or even
    fn hash_to(&self, state: &mut HashStream);
}
```

But given that `HashStream` doesn't work as efficiently as expected, your solution may be appropriate.

Reply:
`StorageValue` requires `UniqueHash`, not `CryptoHash`.